### PR TITLE
Dialogs to functional

### PR DESCRIPTION
**Pull request by flacoman91**
Short description explaining the high-level reason for the pull request

## Additions

- Better unit tests using RTL
- adding missing test coverage for tour and advanced search tips

## Removals

- Modal reducer - moved contents of modal reducer into View since there's only 1 value we use for it.
- showMoreAboutDialog action. This isn't necessary and just duplicates the showModal action
- Remove the download file link logic.  I think this may be some remnants of some IE11 things or code that wasn't supported when this code was written. Removed domUtils since we don't actually need to make a fake link in the dom to trigger a click anymore
- remove the `required` labels from Data Export form since these values are radio buttons and there's no way to have any empty radio buttons
- snapshot testing

## Changes

- Simplified the logic to show modals
- Moved AdvancedTips out of Dialogs since it isn't a modal component
- leverage use of modern react hooks to enable better form validation in the DataExport dialog
- Moved Loading out of Dialogs directory since it's not a modal component
- Moved Export Confirmation into its own component
- removed actions/dataExports into the DataExport components directory since the contents of this file are not redux actions. 

## Testing

- build code
- Click export data
- Download file as csv or json
- do the same again with filtered data
- confirm copy link works

## Notes

-

## Todos

-

## Checklist

* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

File: src/components/Dialogs/DataExport/DataExport.spec.js [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+
+    expect(
+      screen.getByRole('button', { name: 'Start export' })
+    ).toBeInTheDocument();
+    expect(screen.getByRole('button', { name: 'Cancel' })).toBeInTheDocument();
+    const buttonCopy = screen.getByRole('button', { name: 'Copy' });
+    expect(buttonCopy).toBeInTheDocument();
+    expect(buttonCopy).toHaveClass('a-btn__secondary');
+    fireEvent.click(buttonCopy);
+    await (() => {
```
Comment (KatherynRizek) [inline src/components/Dialogs/DataExport/DataExport.spec.js:63]: This should be `await waitFor( () => {...} ); ` The `await` by itself will pass everything

File: src/components/Dialogs/RootModal.spec.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    //   .mockImplementation(() => jest.fn());
+    render(<RootModal />, {
+      preloadedState: {
+        view: {
+          modalTypeShown: types.MODAL_TYPE_EXPORT_CONFIRMATION,
+        },
+      },
+    });
+    expect(screen.getByText('Export complaints')).toBeInTheDocument();
+    // TODO: uncomment this out when we upgrade to react 18.
```
Comment (KatherynRizek) [inline src/components/Dialogs/RootModal.spec.js:50]: Hopefully the upgrade will fix this!

File: src/components/Dialogs/DataExport/DataExport.spec.js (hunk 1 above)
Comment (flacoman91) [reply src/components/Dialogs/DataExport/DataExport.spec.js]: awesome, thanks!

File: src/components/Dialogs/RootModal.spec.js (hunk 2 above)
Comment (flacoman91) [reply src/components/Dialogs/RootModal.spec.js]: Looks like it did.  Good stuff 👍

File: src/components/Dialogs/DataExport/DataExport.spec.js (hunk 1 above)
Comment (flacoman91) [reply src/components/Dialogs/DataExport/DataExport.spec.js]: You're right.  By adding waitFor and also following the eslint nag of not combining the await conditions, an error was caught.

**Review by KatherynRizek — APPROVED** [review]

**Pull request merged**